### ISSUE TITLE
batch_encode_plus returns same lengths when enable pad_to_max_length

### ISSUE THREAD

**Issue by binh-vu** [huggingface/transformers#4471]
# 🐛 Bug

## Information

Model I am using (Bert, XLNet ...): Bert

Language I am using the model on (English, Chinese ...): English

The problem arises when using:
* [ ] the official example scripts: (give details below)
* [x] my own modified scripts: (give details below)

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [ ] my own task or dataset: (give details below)

## To reproduce

1. Run the following script

```
tokenizer = BertTokenizer.from_pretrained("bert-base-uncased")
sents = [
    "I can eat glass without harm",
    "I cannot eat glass"
]
resp = tokenizer.batch_encode_plus(sents, pad_to_max_length=True, return_lengths=True)
print(resp['length'])
# >>> get [8, 8], should be [8, 6]
```
## Expected behavior

The function batch_encode_plus should return correct lengths of sentences before padded to max length. Which should be [8, 6] in the above example. Otherwise, we can just get the length from the last dimension of the mask.

## Environment info

     
- `transformers` version:
- Platform: Ubuntu
- Python version: 3.7
- PyTorch version (GPU?): CPU 1.5
- Tensorflow version (GPU?): No
- Using GPU in script?: No
- Using distributed or parallel set-up in script?: No

**Comment by BramVanroy** [comment]
This is not a bug but expected behaviour. The length of the tokenized input is only calculated after padding.

https://github.com/huggingface/transformers/blob/a08652772791fdaeed6f263b1a99926ca64be5dc/src/transformers/tokenization_utils.py#L1981-L1982

Perhaps you are right, though, and it would be more useful to get the size before padding!
